src/main/java/io/cdap/plugin/format/MultiSQLStatementInputFormat.java

```diff
@@ -62,23 +65,30 @@ public static void setInput(Configuration hConf, MultiTableConf dbConf,
   public List<InputSplit> getSplits(JobContext context) throws IOException {
     MultiTableDBConfiguration conf = new MultiTableDBConfiguration(context.getConfiguration());
 
+    return getSplits(conf);
+  }
+
+  @VisibleForTesting
+  protected List<InputSplit> getSplits(MultiTableDBConfiguration conf) {
     List<String> sqlStatements = conf.getSqlStatements();
     List<String> tableAliases = conf.getTableAliases();
     List<InputSplit> resultSplits = new ArrayList<>();
 
-    if (tableAliases.size() == 0) {
-      tableAliases = new ArrayList<>(sqlStatements.size());
+    //Handle the case where there are not enough table aliases for all SQL statements.
+    //In this case, we use as many aliases as possible and then fall back to the default logiv.
+    if (tableAliases.size() < sqlStatements.size()) {
+      tableAliases = new LinkedList<>(tableAliases);
       while (tableAliases.size() < sqlStatements.size()) {
```

Comment:
You can also use https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#nCopies(int,%20T)